src/core/src/FabricMaterial.cpp

```diff
@@ -62,175 +52,177 @@ void FabricMaterial::setActive(bool active) {
-            srw.setArrayAttributeSize(baseColorTexPathFabric, FabricTokens::_sdrMetadata, 0);
-
-            // clang-format off
-            auto infoImplementationSourceFabric = srw.getAttributeWr<omni::fabric::Token>(baseColorTexPathFabric, FabricTokens::info_implementationSource);
-            auto infoMdlSourceAssetFabric = srw.getAttributeWr<omni::fabric::AssetPath>(baseColorTexPathFabric, FabricTokens::info_mdl_sourceAsset);
-            auto infoMdlSourceAssetSubIdentifierFabric = srw.getAttributeWr<omni::fabric::Token>(baseColorTexPathFabric, FabricTokens::info_mdl_sourceAsset_subIdentifier);
-            auto paramColorSpaceFabric = srw.getArrayAttributeWr<omni::fabric::Token>(baseColorTexPathFabric, FabricTokens::_paramColorSpace);
-            // clang-format on
-
-            *infoImplementationSourceFabric = FabricTokens::sourceAsset;
-            infoMdlSourceAssetFabric->assetPath = UsdTokens::gltf_pbr_mdl;
-            infoMdlSourceAssetFabric->resolvedPath = pxr::TfToken();
-            *infoMdlSourceAssetSubIdentifierFabric = FabricTokens::gltf_texture_lookup;
-            paramColorSpaceFabric[0] = FabricTokens::inputs_texture;
-            paramColorSpaceFabric[1] = FabricTokens::_auto;
-
-            // Create connection from shader to texture.
-            srw.createConnection(
-                shaderPathFabric,
-                FabricTokens::inputs_base_color_texture,
-                omni::fabric::Connection{omni::fabric::PathC(baseColorTexPathFabric), FabricTokens::outputs_out});
-        }
+    attributes.createAttributes(shaderPath);
+
+    srw.setArrayAttributeSize(shaderPath, FabricTokens::_paramColorSpace, 0);
+    srw.setArrayAttributeSize(shaderPath, FabricTokens::_sdrMetadata, 0);
+
+    // clang-format off
+    auto inputsExcludeFromWhiteModeFabric = srw.getAttributeWr<bool>(shaderPath, FabricTokens::inputs_excludeFromWhiteMode);
+    auto infoImplementationSourceFabric = srw.getAttributeWr<omni::fabric::Token>(shaderPath, FabricTokens::info_implementationSource);
+    auto infoMdlSourceAssetFabric = srw.getAttributeWr<omni::fabric::AssetPath>(shaderPath, FabricTokens::info_mdl_sourceAsset);
+    auto infoMdlSourceAssetSubIdentifierFabric = srw.getAttributeWr<omni::fabric::Token>(shaderPath, FabricTokens::info_mdl_sourceAsset_subIdentifier);
+    // clang-format on
+
+    *inputsExcludeFromWhiteModeFabric = false;
+    *infoImplementationSourceFabric = FabricTokens::sourceAsset;
+    infoMdlSourceAssetFabric->assetPath = Context::instance().getCesiumMdlPathToken();
+    infoMdlSourceAssetFabric->resolvedPath = pxr::TfToken();
+    *infoMdlSourceAssetSubIdentifierFabric = FabricTokens::cesium_material;
+
+    if (hasVertexColors) {
+        const auto vertexColorPrimvarNameSize = UsdTokens::vertexColor.GetString().size();
+        srw.setArrayAttributeSize(shaderPath, FabricTokens::inputs_vertex_color_name, vertexColorPrimvarNameSize);
+        auto vertexColorNameFabric =
```

Comment:
What's the use case for a vertex color name?

Reply:
This is the fabric attribute that populates the `vertex_color_name` uniform in `gltf/pbr.mdl`, since we want to override the default vertex color name.

See `extern/nvidia/_build/target-deps/kit-sdk/mdl/core/mdl/gltf/pbr.mdl`